firedrake/adjoint/blocks.py

```diff
@@ -32,8 +31,11 @@ class ConstantAssignBlock(blocks.ConstantAssignBlock, Backend):
 
 class FunctionAssignBlock(blocks.FunctionAssignBlock, Backend):
     def recompute_component(self, inputs, block_variable, idx, prepared):
-        result = super().recompute_component(inputs, block_variable, idx, prepared)
-        return maybe_disk_checkpoint(result)
+        if hasattr(block_variable.checkpoint, "other"):
```

Comment:
Check for being an instance of DelegatedCheckpoint.

Reply:
Done.